tests/layers/tuple_matchers/slot_similarity_tuple_matcher_test.py

```diff
@@ -0,0 +1,114 @@
+# pylint: disable=no-self-use
+from unittest import TestCase
+
+import numpy
+from keras import backend as K
+from keras.layers import Input, Masking
+from keras.models import Model
+from numpy.testing import assert_array_almost_equal
+from scipy.stats import logistic
+
+from deep_qa.layers.tuple_matchers.slot_similarity_tuple_matcher import SlotSimilarityTupleMatcher
+from deep_qa.tensors.similarity_functions.cosine_similarity import CosineSimilarity
+
+
+class TestSlotSimilarityTupleMatcher(TestCase):
+
+    def setUp(self):
+        num_slots = 3
+        embed_dimension = 4
+        self.tuple1_input = Input(shape=(num_slots, embed_dimension), dtype='float32', name="input_tuple1")
+        self.tuple2_input = Input(shape=(num_slots, embed_dimension), dtype='float32', name="input_tuple2")
+        self.num_hidden_layers = 1
+        self.hidden_layer_width = 2
+        self.hidden_layer_activation = 'linear'
+
+        # tuple1 shape: (batch size, num_slots, embed_dimension)
+        self.tuple1 = numpy.asarray([[numpy.random.random(embed_dimension) for _ in range(num_slots)]],
```

Comment:
You can just do `self.tuple1 = numpy.random.rand(1, num_slots, embed_dimension)`